test/gpdb/test_in_docker.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+function install_pgbackrest() {
+    pushd /home/gpadmin/pgbackrest/src
+    ./configure --enable-test &&
+    make -j`nproc` -s &&
+    make install &&
+    make clean
+    popd
+}
+
+function stop_and_clear_gpdb() {
+    source gpdb_src/gpAux/gpdemo/gpdemo-env.sh
+    DATADIR="${MASTER_DATA_DIRECTORY%*/*/*}"
+    # Attempt to stop the Greenplum Database cluster gracefully
+    echo "Attempting to stop GPDB cluster gracefully..."
+    su - gpadmin -c "
+        export MASTER_DATA_DIRECTORY=$MASTER_DATA_DIRECTORY
+        source /usr/local/greenplum-db-devel/greenplum_path.sh &&
+        gpstop -af "
```

Comment:
if gpstop fails, the script will terminate. I think we need to do something like:
```suggestion
    if  ! su - gpadmin -c "
        export MASTER_DATA_DIRECTORY=$MASTER_DATA_DIRECTORY
        source /usr/local/greenplum-db-devel/greenplum_path.sh &&
        gpstop -af "; then

        echo "Graceful stop failed. Attempting to force stop any \
        remaining postgres processes..."
        pkill -9 postgres
    fi
```